SalaryService.Application/Services/CompensationsService.cs

```diff
@@ -51,4 +54,41 @@ public async Task CreateAsync(CompensationCreateDto dto, Employee employee)
     {
         await _compensationCreationCommand.ExecuteAsync(dto, employee);
     }
+
+    public async Task<CompensationCeoListDto> GetAdminAllAsync()
+    {
+        var compensations = await _compensationsQuery.GetCompensationsAsync();
+
+        var compensationCeoList = compensations.Select(x => new CompensationCeoItemDto()
+        {
+            Id = x.Id,
+            EmployeeFullName = x.Employee.GetFullName(),
+            Comment = x.Comment,
+            Amount = x.Amount,
+            IsPaid = x.IsPaid,
+            DateCreateCompensation = x.DateCreateCompensation.ToString(),
+            DateCompensation = x.DateCompensation.ToString()
+        }).ToList();
+
+        var totalAmount = Math.Round(compensations.Sum(x => x.Amount), 2);
+
+        var compensationsResponseList = new CompensationCeoListDto()
+        {
+            List = compensationCeoList,
+            TotalAmount = totalAmount
+        };
+
+        return compensationsResponseList;
+    }
+
+    public async Task UpdateStatusAsync(bool isPaid, long[] compensationsIds)
+    {
+        foreach(var compensationId in compensationsIds)
+        {
+            var compensation = await _compensationsQuery.GetCompensationByIdAsync(compensationId);
```

Comment:
I see a problem here - you don't handle the behavior when a compensation is not found. Please decide what should be in this case